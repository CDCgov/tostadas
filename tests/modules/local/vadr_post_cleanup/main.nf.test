nextflow_process {

    name "Test Process VADR_POST_CLEANUP"
    script "modules/local/vadr_post_cleanup/main.nf"
    process "VADR_POST_CLEANUP"

    test("Should run without failures") {

        when {
            params {
                // define parameters here.
            }
            process {
                """
                input[0] =  [ [ batch_id: "batch_1",  batch_tsv: file("${projectDir}/tests/modules/local/repeatmasker_annotation/batch_1.tsv"), sample_id: "NY0006" ], 
                               file("${projectDir}/tests/modules/local/vadr_post_cleanup/NY0006_mpxv/") 
                            ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
