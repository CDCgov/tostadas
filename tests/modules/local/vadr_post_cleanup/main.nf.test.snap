{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "NY0006_reformatted.gff:md5,a31c1f0f5a4d22f7e74078c3a15a3819"
                    ]
                ],
                "1": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "annotation_error.txt:md5,51bd077e16a642d11d2de8b0973f4115"
                    ]
                ],
                "2": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "NY0006.tbl:md5,eab8a3017b80b54d932cac2c86742438"
                    ]
                ],
                "errors": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "annotation_error.txt:md5,51bd077e16a642d11d2de8b0973f4115"
                    ]
                ],
                "gff": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "NY0006_reformatted.gff:md5,a31c1f0f5a4d22f7e74078c3a15a3819"
                    ]
                ],
                "tbl": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "NY0006.tbl:md5,eab8a3017b80b54d932cac2c86742438"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-06-23T18:52:49.93027507"
    }
}